Time Domain Analysis Functions.R

```diff
@@ -0,0 +1,433 @@
+      
+      time_vector <- append(sumVector, newsum)
+      
+    }
+    
+    seconds_vector <- time_vector / 1000
+    
+    newdata <- cbind(seconds_vector, RR)
+    
+    time_constrained_RRvalues <- newdata[min:max, 2]
+    
+    plot(time_constrained_RRvalues, type = "l", xlab = "Seconds", ylab = "RR Values")
+    
+  }
+  
+  else if (input = "MS" && segment = TRUE) {
+    
+    RR <- diff(data[ , 1])
+    
+    #create an extra column of milliseconds by converting the RR values 
+    
+    time_vector <- vector()
+    
+    for (value in 1:nrow(x)) {
+      
+      newsum <- sum(x[1:value, ])
+      
+      time_vector <- append(sumVector, newsum)
+      
+    }
+    
+    seconds_vector <- time_vector / 1000
+    
+    newdata <- cbind(seconds_vector, RR)
+    
+    time_constrained_RRvalues <- newdata[min:max, 2]
+    
+    plot(time_constrained_RRvalues, type = "l", xlab = "Seconds", ylab = "RR Values")
+    
+  }
+
+}
+
+
+# SDRR vs. SDARR Calculation Function 
+
+# SDRR and SDARR are just a way to discirminate between a segment shorter 
+
+# or longer than 5 min -- because segments longer than 5min are more reliable
+
+# both are the standard deviation of all RR values
+
+
+SdannSdnnFunction <- function(input) {
+  
+if (input = "RR") {
+
+  RR <- data[ , 1]
+  
+  time_vector <- vector()
+  
+  for (value in 1:nrow(x)) {
+    
+    newsum <- sum(x[1:value, ])
+    
+    time_vector <- append(sumVector, newsum)
+    
+  }
+  
+} else if (input = "MS") {
+  
+  time_vector <- data[ ,1]
+  
+}
+  
+  minutes <- time_vector / 60000
+  
+# then goes into determining if it is an SDRR or SDARR
+  
+  if (tail(minutes) < 5) {
+    
+    # use SDNN
+    
+    if (input == "RR") {
+    
+      sdRR <- sd(data[,1])
+    
+      return(sdRR)
+  
+    } else if (input == "MS") {
+    
+      diffRR <- diff(data[,1])
+    
+      sddiffRR <- sd(diffRR)
+    
+      return(sddiffRR)
+      
+    }
+
+  } else if (tail(minutes) >= 5) {
+    
+    # use SDARR
+  
+      if (input == "RR") {
+    
+        sdRR <- sd(data[,1])
+    
+        return(sdRR)
+  
+    } else if (input == "MS") {
+    
+        diffRR <- diff(data[,1])
+    
+        sddiffRR <- sd(diffRR)
+    
+        return(sddiffRR)
+  
+        }
+
+    }
+
+  }
+
+
+# pNN50 Values
+
+# tells us how many RR intervals are larger than 50 
+
+# used more in medical rather than psychological research 
+
+
+pNN50_Function <- function(x, input) {
+
+  if (input = "RR") {
+
+    RR <- data[ , 1]
+
+  }
+
+  if (input = "MS") {
+    
+    RR <- diff(data[ , 1])
+    
+  }
+  
+  counter <- 0
+  
+  for (i in 1:length(RR)) {
+    
+    if(RR > 50) {
+      
+      counter <- counter + 1
+      
+    }
+  
+  }
+  
+}
+
+
+
+
+# Descriptive Heart Rate -- MIN + MAX + AVERAGE
+
+Descriptive_HR_Function <- function (data, input) {
+  
+  if (input = "MS") {
+    
+    RR <- diff(data [ ,1])
+    
+  }
+  
+  else if (input = "RR") {
+    
+    RR <- data[ ,1]
+    
+  }
+  
+  return(mean(RR))
```

Comment:
Great that you implemented this descriptive HR function! While separately returning mean, min and max works, a potential user might be interested in more descriptive metrics. You could use the function describe from the package psych (so psych::describe) that returns  those values and the median, standard error etc. There are many functions like this so you could consider to make use of them :)

Reply:
In addition, if you choose to keep this function, you could also consider returning all values at once. To do this, you can make a list of your values, and return this list, rather than 3 separate values.